src/conn/conn_sweep.c

```diff
@@ -45,6 +81,20 @@ __sweep_mark(WT_SESSION_IMPL *session, uint64_t now)
           __wt_atomic_loadi32(&dhandle->session_inuse) > 0 || dhandle->timeofdeath != 0)
             continue;
 
+        /* For table dhandles, skip expiration if associated file dhandles exist. */
+        if (dhandle->type == WT_DHANDLE_TYPE_TABLE) {
+            WT_WITHOUT_DHANDLE(session,
+              WT_WITH_HANDLE_LIST_READ_LOCK(
```

Comment:
Should we just onLy parse through simple file handles? And when we expire the file handle, we follow expire the table handle as well? Wouldn't that solve this problem as well?